java/hive/src/main/java/io/v6d/hadoop/fs/FileSystem.java

```diff
@@ -378,18 +380,54 @@ public boolean renameInternal(Path path, Path path1) throws IOException {
             Files.delete(nioFilePath);
         } else {
             Files.move(nioFilePath, nioFilePath1);
+            ByteBuffer bytes = ByteBuffer.allocate(255);
+            FileChannel channel = FileChannel.open(nioFilePath1, StandardOpenOption.READ);
+            int len = channel.read(bytes);
+            String objectIDStr =
+                    new String(bytes.array(), 0, len, StandardCharsets.UTF_8).replaceAll("\n", "");
+            IPCClient client = Context.getClient();
+            client.putName(ObjectID.fromString(objectIDStr), nioFilePath1.toString());
+            try {
+                client.dropName(nioFilePath.toString());
+            } catch (Exception e) {
+                Context.println("Drop name error: " + e.getMessage());
+            }
         }
         printAllFiles();
 
         return true;
     }
 
+    private void syncWithVineyard(String prefix) throws IOException {
+        IPCClient client = Context.getClient();
+        try {
+            String reg = "^" + prefix + ".*";
+            Map<String, ObjectID> objects = client.listNames(reg, true, 255);
+            for (val object : objects.entrySet()) {
+                if (Files.exists(jimfs.getPath(object.getKey()))) {
+                    continue;
+                }
+                Files.createFile(jimfs.getPath(object.getKey()));
+                FileChannel channel =
+                        FileChannel.open(jimfs.getPath(object.getKey()), StandardOpenOption.WRITE);
+                ObjectID id = object.getValue();
+                channel.write(
+                        ByteBuffer.wrap((id.toString() + "\n").getBytes(StandardCharsets.UTF_8)));
+                channel.close();
+            }
+        } catch (Exception e) {
+            Context.println("Exception: " + e.getMessage());
+        }
+    }
+
     @Override
     public FileStatus[] listStatus(Path path) throws FileNotFoundException, IOException {
+        Context.println("listStatus:" + path.toString());
```

Comment:
I think this line is only for debugging?

Reply:
yep. I've cleared it.